### PR TITLE
Add support for RBAC in weaviate-cli.

### PR DESCRIPTION
**Pull request by jfrancoa**
This commit adds the option to add, remove, get roles. Assigning roles to users, as well as permissions to roles, revoking them (roles as well as permissions).

It also adds the option to specify a user from your config as the user to pass in the client connection. This allows having multiple api-keys assigned to different users without having the need to keep different configs.

### CODE REVIEW

File: weaviate_cli/commands/get.py [hunk 1]
```diff
@@ -141,3 +149,103 @@ def get_backup_cli(ctx, backend, backup_id, restore):
...
+        if client:
+            client.close()
+        sys.exit(1)
+    finally:
+        if client:
+            client.close()
+
+
+@get.command("user")
+# @click.option(
```
Comment (rlmanrique) [inline weaviate_cli/commands/get.py:213]: I'd delete the comments

File: weaviate_cli/commands/get.py [hunk 2]
```diff
@@ -141,3 +149,103 @@ def get_backup_cli(ctx, backend, backup_id, restore):
...
+#     "--user",
+#     default=GetUserDefaults.user_id,
+#     help="The ID of the user to get.",
+# )
+@click.option(
+    "--role_name",
+    default=GetUserDefaults.role_name,
+    help="The name of the role to get the users assigned to.",
+)
+# @click.option(
```
Comment (rlmanrique) [inline weaviate_cli/commands/get.py:223]: I'd delete the comments

File: weaviate_cli/commands/revoke.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import click
```
Comment (rlmanrique) [inline weaviate_cli/commands/revoke.py:1]: Aa a minor comment, I am not sure is needed a revoke command. I know there is a nuance with revoke and delete, but to me, it'd be more intuitive to have it under delete.

Just an option, no change needed if you don't agree :)

File: weaviate_cli/commands/revoke.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            client.close()
+
+
+@revoke.command("permission")
+@click.option(
+    "-p",
+    "--permission",
+    multiple=True,
+    required=True,
+    help=PERMISSION_HELP_STRING,
```
Comment (rlmanrique) [inline weaviate_cli/commands/revoke.py:51]: Could this message be more descriptive? with an example maybe?

File: weaviate_cli/defaults.py [hunk 5]
```diff
@@ -1,5 +1,27 @@
-from dataclasses import dataclass
-from typing import Optional
+from dataclasses import dataclass, field
+from typing import Optional, List, Dict
+
+
+PERMISSION_HELP_STRING = (
```
Comment (rlmanrique) [inline weaviate_cli/defaults.py:5]: Ah ok. I saw it here. Forget my previous comment

**Review by rlmanrique — APPROVED** [review]

File: weaviate_cli/commands/revoke.py (hunk 3 above)
Comment (jfrancoa) [reply weaviate_cli/commands/revoke.py:1]: This is something I was considering at the beginning, however I wanted to keep it consistent with the roles logic. A role can be assigned to a user, as a permission can be assigned to a role. The creation and deletion of the roles is handled by create and delete, but for the permissions there are no options to create them, they are pre-defined.
What I have done has been modifying the `add` and rename it by `assign`. After reading your comment I think it makes more sense.

**Pull request merged**